### PR TITLE
feat: Add Chicken In/Out trait probabilities

### PR DESCRIPTION
**Pull request by bingen**
:warning: TODO: Change base to main!

### CODE REVIEW

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol [hunk 1]
```diff
@@ -0,0 +1,130 @@
+pragma solidity ^0.8.11;
+
+import "./EggTraitWeights.sol";
+
+
+contract ChickenInTraitWeights is EggTraitWeights {
+    uint256 constant EGG_CHICKEN_REPEAT_FACTOR = 120e16;
+    // See: https://www.desmos.com/calculator/fctburkmtg
+    uint256 constant R3 = 52e16;
+    uint256 constant ALPHA3 = 1e18 / 3;
```
Comment (danielattilasimon) [inline LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:10]: 3 should be explicitly converted to `uint256`

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        if (rand < needle) { return ShellColor.Gold; }
+        needle += chickenWeightsCached[uint256(ShellColor.Rainbow)];
+        if (rand < needle) { return ShellColor.Rainbow; }
+        return ShellColor.Luminous;
+    }
+
+    function _getChickenTrait9(uint256 rand, uint256 troveFactor) internal view returns (uint8) {
+        // TODO: should we hardcode this? Gas doesn’t really matter here as we are just static-calling them
+        uint256 tier1Probability = ((2e18 - ALPHA3) / 3 - R3 * troveFactor / 1e18) / 3;
+        uint256 tier2Probability = 1e18 / 9;
```
Comment (danielattilasimon) [inline LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:74]: Same.

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        if (rand < needle) { return 2; }
+        needle += tier3Probability;
+        if (rand < needle) { return 3; }
+        return 4;
+    }
+
+    function _getChickenWing(uint256 rand, uint256 troveFactor) internal view returns (uint8) {
+
+        uint256 tier1Probability = (2e18 - ALPHA3) / 3 - R3 * troveFactor / 1e18;
+        uint256 tier2Probability = 1e18 / 3;
```
Comment (danielattilasimon) [inline LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:121]: Same.

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        uint256 tier3Probability = ALPHA3 / 3 + R3 * troveFactor / 1e18;
```
Comment (danielattilasimon) [inline LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:122]: `tier3Probability` is unused.

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    function _getChickenTrait9(uint256 rand, uint256 troveFactor) internal view returns (uint8) {
+        // TODO: should we hardcode this? Gas doesn’t really matter here as we are just static-calling them
+        uint256 tier1Probability = ((2e18 - ALPHA3) / 3 - R3 * troveFactor / 1e18) / 3;
+        uint256 tier2Probability = 1e18 / 9;
+        uint256 tier3Probability = (ALPHA3 / 3 + R3 * troveFactor / 1e18) / 3;
+
+        uint256 needle = tier1Probability;
+        if (rand < needle) { return 1; }
```
Comment (danielattilasimon) [inline LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:78]: Great that we're using 1-based indices! Makes it simpler to convert it to metadata.

File: LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol (hunk 1 above)
Comment (bingen) [reply LUSDChickenBonds/src/NFTArtwork/ChickenInTraitWeights.sol:10]: Sorry, it wasn’t even compiling! :sweat_smile:

**Pull request merged**